packages/forms/examples/file-upload.md

```diff
@@ -0,0 +1,161 @@
+The `FileUpload` component is a styled `<div>` which can be used
+with 3rd-party file upload libraries like [react-dropzone](https://github.com/react-dropzone/react-dropzone/).
+
+The example below includes an a mock file upload implementation that
+accepts any file type or size.
+
+```jsx
+const { useDropzone } = require('react-dropzone');
+const { Well } = require('@zendeskgarden/react-notifications/src');
+const { Code } = require('@zendeskgarden/react-typography/src');
+const { getColor } = require('@zendeskgarden/react-theming/src');
+const { IconButton } = require('@zendeskgarden/react-buttons/src');
+const { Progress } = require('@zendeskgarden/react-loaders/src');
+const FileImageStroke = require('@zendeskgarden/svg-icons/src/16/file-image-stroke.svg').default;
+const CloseStroke = require('@zendeskgarden/svg-icons/src/16/x-stroke.svg').default;
+
+const StyledFileWrapper = styled.ul`
+  direction: ${p => (p.theme.rtl ? 'rtl' : 'ltr')};
+  margin-top: ${p => p.theme.space.sm};
+  list-style: none;
+  padding: 0;
+
+  & > *:not(:first-child) {
+    margin-top: ${p => p.theme.space.xs};
+  }
+`;
+
+const StyledFile = styled.div`
+  position: relative;
+  border: ${p => p.theme.borders.sm};
+  border-color: ${p => getColor('neutralHue', 300, p.theme)};
+  border-radius: ${p => p.theme.borderRadii.md};
+  padding: ${p => p.theme.space.xxs} ${p => p.theme.space.xs};
+  display: inline-flex;
+  min-width: 200px;
+  flex-wrap: nowrap;
+`;
+
+const StyledFileName = styled.span`
+  display: flex;
+  align-items: center;
+  flex-grow: 1;
+  padding: 0 ${p => p.theme.space.xs};
+`;
+
+const StyledFileIcon = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const StyledProgress = styled(Progress)`
+  position: absolute;
+  left: 0;
+  bottom: 0;
+  right: 0;
+  margin: 0;
+`;
+
+const File = React.memo(({ name, onRemove }) => {
+  const [uploadProgress, setUploadProgress] = React.useState(0);
+
+  React.useEffect(() => {
+    const randomUploadInterval = Math.random() * (600 - 200) + 600;
+
+    const uploadInterval = setInterval(() => {
+      setUploadProgress(prevProgress => {
+        if (prevProgress >= 100) {
+          clearInterval(uploadInterval);
+          return 100;
+        }
+
+        return prevProgress + 10;
+      });
+    }, randomUploadInterval);
+
+    return () => {
+      clearInterval(uploadInterval);
+    };
+  }, []);
+
+  return (
+    <StyledFile aria-label="File">
+      <StyledFileIcon>
+        <FileImageStroke />
+      </StyledFileIcon>
+      <StyledFileName aria-label="File name">{name}</StyledFileName>
+      <IconButton size="small" onClick={onRemove} aria-label="Remove file">
+        <CloseStroke />
+      </IconButton>
```

Comment:
[nit] given the constrained space, these might benefit from `focusInset`. What do you think, @steue?

Reply:
Actually, we already know from the overflow button treatment on https://zendeskgarden.github.io/react-components/tables/#overflow-menu that this should be `focusInset`.

Reply:
Updated.